### PR TITLE
Enable rounding for gas

### PR DESCRIPTION
**Pull request by dariorussi**
## Description 

Actually enable rounding

## Test Plan 

current tests

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/gas_model/gas_v2.rs [hunk 1]
```diff
@@ -326,11 +326,11 @@ impl SuiGasStatusAPI for SuiGasStatus {
 
     fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
         let gas_used = self.gas_status.gas_used_pre_gas_price();
-        let gas_used = if self.gas_status.gas_model_version > 5 {
-            if gas_used > 0 && gas_used % self.gas_rounding_step == 0 {
+        let gas_used = if let Some(gas_rounding) = self.gas_rounding_step {
```
Comment (tzakian) [inline crates/sui-types/src/gas_model/gas_v2.rs:329]: Nice!

**Review by tzakian — APPROVED** [review]
LGTM!

**Comment by dariorussi** [comment]
@tzakian @oxade thoughts around the size limit tests changes?

**Comment by oxade** [comment]
> @tzakian @oxade thoughts around the size limit tests changes?

All clear here :)

**Pull request merged**